marble-table/src/test/java/org/apache/calcite/table/TypeConvertUtilTest.java

```diff
@@ -0,0 +1,158 @@
+package org.apache.calcite.table;
+
+import org.apache.calcite.util.NlsString;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TypeConvertUtilTest {
+
+  @Test
+  public void testCastBigDecimal() {
+    final BigDecimal actual =
+            TypeConvertUtil.castBigDecimal(new BigDecimal(5));
+    Assert.assertEquals(new BigDecimal(5), actual);
+  }
+
+  @Test
+  public void testCastBoolean() {
+    Assert.assertFalse(TypeConvertUtil.castBoolean(false));
+  }
+
+  @Test
+  public void testCastDouble() {
+    Assert.assertEquals(3.0, TypeConvertUtil.castDouble(3.0), 0.0);
+  }
+
+  @Test
+  public void testCastFloat() {
+    Assert.assertEquals(4.0f, TypeConvertUtil.castFloat(4.0f), 0.0);
+  }
+
+  @Test
+  public void testCastInt() {
+    Assert.assertEquals(new Integer(0), TypeConvertUtil.castInt(0));
+  }
+
+  @Test
+  public void testCastList() {
+    final ArrayList value = new ArrayList();
+    value.add(6);
+    final List actual = TypeConvertUtil.castList(value);
+
+    final ArrayList arrayList = new ArrayList();
```

Comment:
I am quite sure an ` java.util.ArrayList` is defined as `ArrayList<Something> name = new ArrayList<>()`
same with the list

Reply:
Done